File: python/potega.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
#  potega.py


def potega_it(x, n):
    i = 1
    wynik = 1
    while i <= n:
        wynik = wynik * x
        i = i + 1
    return wynik


# potega_rek(x, 0) = 1 dla x różnego od 0
# potega_rek(x, n) = potega_rek(x, n-1) * x dla n = N+
def potega_rek(x, n):
    if x == 0:
        return 0
    elif n == 0:
        return 1
    return potega_rek(x, n - 1) * x


def main(args):
    x = float(input("Podaj podstawe potęgi: "))
    n = int(input("Podaj wykładnik potęgi: "))
    assert type(n) == int
    assert potega_it(1, 1) == 1
    assert potega_it(2, 1) == 2
    assert potega_it(2, 3) == 8
    assert potega_it(5, 2) == 25
    print('Wynik: ', potega_rek(x, n))
    return 0


if __name__ == '__main__':
    import sys
sys.exit(main(sys.argv))

File: cpp/potega.cpp
/*
 * potega.cpp
 * 
 * a0 = 1
 * a1 = a
 * an = a * ... * a (n-czynników) dla n zaw. N+ - {1}
 */


#include <iostream>

using namespace std;

float potega_it(float x, int n)
{
    float wynik = 1;
    for (int i = 1; i <= n; i++)
    {
        wynik = wynik * x;
    }
    return wynik;
}

int main(int argc, char **argv)
{
    int n;
    float x;
    cout << "Podaj podstawe potęgi: " << endl;
    cin >> x;
    cout << "Podaj wykładnik potęgi: " << endl;
    cin >> n;
    cout << "Potęga: " << potega_it(x, n) <<endl;
	return 0;
}

